### DISCUSSION TITLE
Receice MAC command

### DISCUSSION THREAD

**Discussion by daeynasvistas** [ngraziano/LMICPP-Arduino#27] [Q&A] [answered]
i do receive MAC command from NS server

![image](https://user-images.githubusercontent.com/2634610/169069265-2599ebaf-33a3-40c2-8d03-d55c9507af35.png)

do the lib really apply the "power to 2" change on the node?
power to 2 are the same as LMIC.setAntennaPowerAdjustment(-14); ?

`ADR REQ Change dr to 5, power to 2`

im very close to the gateway
![image](https://user-images.githubusercontent.com/2634610/169070051-486e3377-1afd-40a8-8cf0-371bd70d16c9.png)

**Comment by daeynasvistas** [comment]
![image](https://user-images.githubusercontent.com/2634610/169072824-a084282b-6513-416f-b3c5-946aea34e483.png)

it seems that its try to change sx1276 power. 
i will try to see the gateway log to compare the dB in there

**Comment by ngraziano** [comment, marked as answer]
Yes the transmitting power is reduce when the network send this type of MAC command.

For example for EU868 the transmission power is -2*the value send by the server (value define in lorawan regional settings).

 So the lib ask to sx1276 to transmit with power :
 14 - 2*(the value in ADR request)  + AntennaPowerAdjustment . 

But the sx1267 power can be set from +2 to +17, so the minimum TX power asked is always +2 (most board do use PA pin see PA_BOOST_PIN)
